tdrs-backend/tdpservice/users/permissions.py

```diff
@@ -61,27 +57,29 @@ def has_permission(self, request, view):
         return is_in_group(request.user, "Data Prepper")
 
 
-class CanDownloadReport(permissions.BasePermission):
-    """Permission for report download."""
-
-    def has_permission(self, request, view):
-        """Check if a user can download file."""
-        is_ofa_admin = bool(
-            is_in_group(request.user, 'OFA Admin') and view.kwargs.get('stt')
-        )
-        return is_ofa_admin or is_own_stt(request, view)
-
-
-class CanUploadReport(permissions.BasePermission):
-    """Permission for report uploads."""
+class ReportFilePermissions(permissions.BasePermission):
+    """Permission for report downloads & uploads."""
 
     def has_permission(self, request, view):
-        """
-        Check if a user is a data prepper or an admin.
+        """Check if a user is a data prepper or an admin.
 
         If they are a data prepper, ensures the STT is their own.
         """
         return (
             is_in_group(request.user, "OFA Admin") or
             is_own_stt(request, view)
         )
+
+    def has_object_permission(self, request, view, obj):
```

Comment:
Unit tests revealed a flaw in the pre-existing permissions classes which resulted in Data Prepper users being able to download files from other STTs. Previously, a Data Prepper's authorization to view a report was determined based on matching the user's STT with the one provided in the request. However, since the download endpoint only accepts the ID of the report we don't have that information available and the previous logic treated no STT in the request as if that would always be allowed.

This leverages built in Django REST Framework capabilities to compare the model instance being requested against the user's set STT to confirm they are the same before allowing a report download.